articles/active-directory/develop/tutorial-v2-windows-uwp.md

```diff
@@ -108,92 +111,95 @@ This section shows how to use MSAL to get a token for Microsoft Graph API. Make
+            IAccount firstAccount = accounts.FirstOrDefault();
 
-         try
-         {
-          authResult = await PublicClientApp.AcquireTokenSilent(scopes, firstAccount)
+            try
+            {
+                authResult = await PublicClientApp.AcquireTokenSilent(scopes, firstAccount)
                                                   .ExecuteAsync();
-         }
-         catch (MsalUiRequiredException ex)
-         {
-          // A MsalUiRequiredException happened on AcquireTokenSilent.
-          // This indicates you need to call AcquireTokenInteractive to acquire a token
-          System.Diagnostics.Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
-
-          try
-          {
-           authResult = await PublicClientApp.AcquireTokenInteractive(scopes)
+            }
+            catch (MsalUiRequiredException ex)
+            {
+                // A MsalUiRequiredException happened on AcquireTokenSilent. 
+                // This indicates you need to call AcquireTokenInteractive to acquire a token
+                System.Diagnostics.Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
+
+                try
+                {
+                    authResult = await PublicClientApp.AcquireTokenInteractive(scopes)
                                                       .ExecuteAsync()
                                                       .ConfigureAwait(false);
-           }
-           catch (MsalException msalex)
-           {
-            await DisplayMessageAsync($"Error Acquiring Token:{System.Environment.NewLine}{msalex}");
-           }
-          }
-          catch (Exception ex)
-          {
-           await DisplayMessageAsync($"Error Acquiring Token Silently:{System.Environment.NewLine}{ex}");
-           return;
-          }
-
-          if (authResult != null)
-          {
-           var content = await GetHttpContentWithToken(graphAPIEndpoint,
-                                                       authResult.AccessToken).ConfigureAwait(false);
-
-           // Go back to the UI thread to make changes to the UI
-           await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
-           {
-            ResultText.Text = content;
-            DisplayBasicTokenInfo(authResult);
-            this.SignOutButton.Visibility = Visibility.Visible;
-           });
-          }
+                }
+                catch (MsalException msalex)
+                {
+                    await DisplayMessageAsync($"Error Acquiring Token:{System.Environment.NewLine}{msalex}");
+                }
+            }
+            catch (Exception ex)
+            {
+                await DisplayMessageAsync($"Error Acquiring Token Silently:{System.Environment.NewLine}{ex}");
+                return;
+            }
+
+            if (authResult != null)
+            {
+                var graphServiceClient = GetGraphServiceClient(authResult.AccessToken);
```

Comment:
No that's not the pattern that the Graph SDK wants us to use.
They want us to use an authentication provider (https://docs.microsoft.com/en-us/graph/sdks/choose-authentication-providers?tabs=CS), or the Azure SDK.

Also if we were to implement the IAuthenticationProvider interface (which might be the thing to do in that tutorial ... although ... why?) ... we'd want the try {silent} catch {MsalUiRequiredException} to be in that implementation. NOT outside.

Reply:
Updated the code as per Public client graph implementation.
Thanks!